crates/oxc_linter/src/rules/jsx_a11y/role_support_aria_props.rs

```diff
@@ -0,0 +1,1601 @@
+    ///     <li tabIndex="-1" role="radio" aria-checked="false">Rainbow Trout</li>
+    ///     <li tabIndex="-1" role="radio" aria-checked="false">Brook Trout</li>
+    ///     <li tabIndex="0" role="radio" aria-checked="true">Lake Trout</li>
+    /// </ul>
+    ///
+    /// // Bad
+    /// <ul role="radiogroup" "aria-labelledby"="foo">
+    ///     <li aria-required tabIndex="-1" role="radio" aria-checked="false">Rainbow Trout</li>
+    ///     <li aria-required tabIndex="-1" role="radio" aria-checked="false">Brook Trout</li>
+    ///     <li aria-required tabIndex="0" role="radio" aria-checked="true">Lake Trout</li>
+    /// </ul>
+    /// ```
+    ///
+    RoleSupportAriaProps,
+    correctness
+);
+
+#[derive(Debug, Default, Clone)]
+pub struct RoleSupportAriaProps;
+
+// TODO: add error message
+#[derive(Debug, Error, Diagnostic)]
+enum RoleSupportAriaPropsDiagnostic {
+    #[error("eslint-plugin-jsx-a11y(role-support-aria-props): The attribute {1} is not supported by the role {2}.")]
+    #[diagnostic(severity(warning), help("Try to remove invalid attribute {1}."))]
+    Default(#[label] Span, String, String),
+
+    #[error("eslint-plugin-jsx-a11y(role-support-aria-props): The attribute {1} is not supported by the role {2}. This role is implicit on the element {3}.")]
+    #[diagnostic(severity(warning), help("Try to remove invalid attribute {1}."))]
+    IsImplicit(#[label] Span, String, String, String),
+}
+
+impl Rule for RoleSupportAriaProps {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::JSXOpeningElement(jsx_el) = node.kind() {
+            if let Some(el_type) = get_element_type(ctx, jsx_el) {
+                let role = has_jsx_prop_lowercase(jsx_el, "role");
+                let role_value = role.map_or_else(
+                    || get_implicit_role(jsx_el, el_type.as_str()),
+                    |i| get_string_literal_prop_value(i),
+                );
+                let is_implicit = role_value.is_some() && role.is_none();
+                if let Some(role_value) = role_value {
+                    if !VALID_ARIA_ROLES.contains(role_value) {
+                        return;
+                    }
+                    let invalid_props = get_invalid_aria_props_for_role(role_value);
+                    for attr in &jsx_el.attributes {
+                        if let JSXAttributeItem::Attribute(attr) = attr {
+                            let name = get_attribute_name(&attr.name).to_lowercase();
+                            if invalid_props.contains(&&name.as_str()) {
+                                ctx.diagnostic(if is_implicit {
+                                    RoleSupportAriaPropsDiagnostic::IsImplicit(
+                                        attr.span,
+                                        name,
+                                        role_value.to_string(),
+                                        el_type.clone(), // TODO: remove clone
```

Comment:
Is it appropriate to .clone here... I feel that it could affect performance and I am using it as a TODO.

Reply:
ah right yeah that's fine here.

ideally it would just be an `Atom` and we'd clone that (it's cheap to clone). but because `el_type` comes from `get_element_type` which possibly reliese on the lint rule config, i don't think it can be

Reply:
I see. Then we will remove the TODO.